components/scream/src/share/field/field_property_checks/field_positivity_check.hpp

```diff
@@ -1,196 +1,26 @@
 #ifndef SCREAM_FIELD_POSITIVITY_CHECK_HPP
 #define SCREAM_FIELD_POSITIVITY_CHECK_HPP
 
-#include "share/field/field_property_check.hpp"
-#include "share/field/field.hpp"
-#include "share/field/field_utils.hpp"
-
-#include "ekat/util/ekat_math_utils.hpp"
+#include "share/field/field_property_checks/field_lower_bound_check.hpp"
 
 namespace scream
 {
 
-// This field property check returns true if all data in a given field are
-// positive, and false if not. If constructed with a specific lower bound, it
-// can repair a field that fails the check by setting all nonpositive values
-// to that lower bound. If no specific lower bound is given (i.e. if the
-// default constructor is used), no repairs can be made.
-template<typename RealType>
-class FieldPositivityCheck: public FieldPropertyCheck<RealType> {
+// Convenience implementation of lower bound check for L=0. The class
+// inherits from FieldLowerBound, and sets L to machine epsilon
+
+class FieldPositivityCheck: public FieldLowerBoundCheck {
 public:
-  using non_const_RT = typename FieldPropertyCheck<RealType>::non_const_RT;
-  using const_RT     = typename FieldPropertyCheck<RealType>::const_RT;
-  using kt = typename Field<RealType>::kt;
-  using exe_space = typename kt::ExeSpace;
-  using RangePolicy = typename kt::RangePolicy;
 
   // Default constructor -- cannot repair fields that fail the check.
-  FieldPositivityCheck () : m_lower_bound(-1) {}
-
-  // Constructor with lower bound -- can repair fields that fail the check
-  // by overwriting nonpositive values with the given lower bound.
-  explicit FieldPositivityCheck (const_RT lower_bound) :
-    m_lower_bound(lower_bound) {
-    EKAT_ASSERT_MSG(lower_bound >= 0, "lower_bound must be positive.");
+  explicit FieldPositivityCheck (const bool can_repair = true) :
+    FieldLowerBoundCheck (std::numeric_limits<double>::epsilon(),can_repair)
```

Comment:
Consider `Real` instead of `double` here (unless we are sticking with double precision)

Reply:
As I said in the other comment, the FPC could be used on a double field (even if Real=float). For that case, a lower bound of `std::numeric_limits<Real>::epsilon()` would be too large.